Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -47,10 +61,11 @@ class OpenSearchEngine {
 
     private func getURLFromTemplate(searchTemplate: String, query: String) -> NSURL? {
         let allowedCharacters = NSCharacterSet(charactersInString: SearchTermsAllowedCharacters)
-
         if let escapedQuery = query.stringByAddingPercentEncodingWithAllowedCharacters(allowedCharacters) {
             let urlString = searchTemplate.stringByReplacingOccurrencesOfString("{searchTerms}", withString: escapedQuery, options: NSStringCompareOptions.LiteralSearch, range: nil)
-            return NSURL(string: urlString)
+            if let encodedUrlString = urlString.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.symbollessURLAllowedSet()) {
```

Comment:
Also, FWIW, we've solved this in the past by just dumping all allowed URL characters: https://github.com/mozilla/firefox-ios/pull/166/files#diff-108efb951993bfa3f37b030dccb77ab8R21. That was eventually backed out, though.

I really wish there was just a `URLAllowedCharacterSet` that just included everything :\ Maybe it'd be worth putting your set into an extension so we don't need to reinvent this everywhere.